diff/diff.go

```diff
@@ -365,7 +366,26 @@ type Do func(a Event) (crud.Arg, error)
 
 func (sc *Syncer) eventLoop(d Do) error {
 	for event := range sc.eventChan {
-		err := sc.handleEvent(d, event)
+		// Stop if program is terminated
+		select {
+		case <-sc.stopChan:
+			return nil
+		default:
+		}
+
+		// For various reasons, Kong can temporarily fail to process
+		// a valid request (e.g. when the database is under heavy load).
+		// We retry each request up to 3 times on failure, after around
+		// 1 second, 3 seconds, and 9 seconds (randomized exponential backoff).
+		exponentialBackoff := backoff.NewExponentialBackOff()
+		exponentialBackoff.InitialInterval = 1 * time.Second
+		exponentialBackoff.Multiplier = 3
+		backoffObject := backoff.WithMaxRetries(exponentialBackoff, 4)
+
+		err := backoff.Retry(func() error {
+			return sc.handleEvent(d, event)
```

Comment:
Can we please emit a log line when an event is being retried? That would help quite a bit with debugging.

Reply:
I agree. However, I don't see any debug logging facility in decK? We have the print package with things like CreatePrintln/DeletePrintln, but those will always print. I imagine that ideally this kind of debug information could be turned on or off, e.g. with a verbose or debug flag.

Reply:
There is a verbose flag in decK. You are right that the machinery around debug logging is lacking and there is not much that can be done easily. Tracking in #248 and #249.